storm-core/src/jvm/org/apache/storm/trident/TridentTopology.java

```diff
@@ -394,11 +395,28 @@ public StormTopology build() {
         Map<Node, String> spoutIds = genSpoutIds(spoutNodes);
         Map<Group, String> boltIds = genBoltIds(mergedGroups);
         
+        Map defaults = Utils.readDefaultConfig();
+
         for(SpoutNode sn: spoutNodes) {
             Integer parallelism = parallelisms.get(grouper.nodeGroup(sn));
+
+            Map<String, Number> spoutRes = null;
+            if(sn instanceof ITridentResource) {
+                spoutRes = mergeDefaultResources(((ITridentResource)sn).getResources(), defaults);
+            }
+            else {
+                spoutRes = mergeDefaultResources(null, defaults);
+            }
```

Comment:
sn is always an instance of ITridentResource and you can have simple null check over here. You may want to replace if/else block with the below.
``` java
            Map<String, Number> spoutRes = mergeDefaultResources((sn != null ? sn.getResources() : null), defaults);
```